src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -78,8 +78,46 @@ public BatchRequestContent(params BatchRequestStep[] batchRequestSteps)
         /// <returns>True or false based on addition or not addition of the provided <see cref="BatchRequestStep"/>. </returns>
         public bool AddBatchRequestStep(BatchRequestStep batchRequestStep)
         {
-            if (batchRequestStep == null || BatchRequestSteps.ContainsKey(batchRequestStep.RequestId))
+            if (batchRequestStep == null
+                || BatchRequestSteps.ContainsKey(batchRequestStep.RequestId)
+                || BatchRequestSteps.Count >= CoreConstants.BatchRequest.MaxNumberOfRequests //we should not add any more steps
+                )
+            {
+                return false;
+            }
+
+            (BatchRequestSteps as IDictionary<string, BatchRequestStep>).Add(batchRequestStep.RequestId, batchRequestStep);
+            return true;
+        }
+
+        /// <summary>
+        /// Adds a <see cref="HttpRequestMessage"/> to batch request content.
+        /// </summary>
+        /// <param name="httpRequestMessage">A <see cref="HttpRequestMessage"/> to use to build a <see cref="BatchRequestStep"/> to add.</param>
+        /// <returns>True or false based on addition or not addition of the provided <see cref="HttpRequestMessage"/>. </returns>
+        public bool AddBatchRequestStep(HttpRequestMessage httpRequestMessage)
```

Comment:
I'm tempted to suggest we return void here instead of bool.  Returning bool significantly complicates the calling code.

Reply:
Since we are moving away from returning bool, I see no issue with returning void and throwing exceptions if MaxNumberOfRequests are exceeded for the new overloads. The intended caller of the new overloads would probably be having a try catch block for graphServiceClient calls. Consistency will be lost, but breaking changes for our customers will be limited to the `AddBatchRequestStep(BatchRequestStep batchRequestStep)` overload.
/cc @darrelmiller, @MIchaelMainer, @andrueastman

Reply:
Also, BatchRequestContent ctor already throws a ClientException with MaximumValueExceeded code.

Reply:
This makes sense to me. 
I will resolve this by returning void in the new overloads, and leave the changing of the existing `AddBatchRequestStep(BatchRequestStep batchRequestStep)` overload be be sorted out in the next time we do a breaking change.

Reply:
This should be resolved now.